diff -up Singular-3-1-6/doc/Makefile.in.destdir Singular-3-1-6/doc/Makefile.in
--- Singular-3-1-6/doc/Makefile.in.destdir	2012-12-19 15:01:16.000000000 -0600
+++ Singular-3-1-6/doc/Makefile.in	2014-03-10 09:48:46.353968370 -0500
@@ -68,7 +68,7 @@ EX_SUBDIR       = ./examples
 CHKSUM_DB       = ${DOC_SUBDIR}/chksum
 DOC2TEX_EXAMPLE_EXCLUSIONS = @DOC2TEX_EXAMPLE_EXCLUSIONS@
 DOC2TEX         = ${PERL} ./doc2tex.pl -docdir ${DOC_SUBDIR} \
-                  -Singular ${SINGULAR} -verbose ${VERBOSE} -make ${MAKE} \
+                  -Singular $(DESTDIR)${SINGULAR} -verbose ${VERBOSE} -make ${MAKE} \
 		  -exdir ${EX_SUBDIR} ${DOC2TEX_EXAMPLE_EXCLUSIONS}
 TAG		=
 PL2DOC		= ${PERL} ./pl2doc.pl ${TAG} -db ${CHKSUM_DB}
diff -up Singular-3-1-6/factory/GNUmakefile.in.destdir Singular-3-1-6/factory/GNUmakefile.in
--- Singular-3-1-6/factory/GNUmakefile.in.destdir	2012-12-19 15:01:16.000000000 -0600
+++ Singular-3-1-6/factory/GNUmakefile.in	2014-03-10 09:48:46.354968360 -0500
@@ -530,43 +530,43 @@ installtest:	test_install
 install install-libsingular:	$(installtargets) cf_gmp.h
 
 install_all:    install libsingcf_p.a 
-		-$(INSTALL_DATA) libsingcf_p.a $(libdir)/libsingcf_p.a 
+		-$(INSTALL_DATA) libsingcf_p.a $(DESTDIR)$(libdir)/libsingcf_p.a 
 
 install-nolns: install
 
 installcf:	cf
-		$(MKINSTALLDIRS) $(libdir)
-		$(MKINSTALLDIRS) $(includedir)
-		$(INSTALL_DATA) $(libfactory) $(libdir)/$(libfactory)
-		-$(INSTALL_DATA) libsingcf_g.a $(libdir)/libsingcf_g.a
-		-$(INSTALL_DATA) libsingcf_p.a $(libdir)/libsingcf_p.a
-		$(INSTALL_DATA) factory.h $(includedir)/factory.h
-		$(INSTALL_DATA) cf_gmp.h $(includedir)/cf_gmp.h
-		$(INSTALL_DATA) factoryconf.h $(includedir)/factoryconf.h
+		$(MKINSTALLDIRS) $(DESTDIR)$(libdir)
+		$(MKINSTALLDIRS) $(DESTDIR)$(includedir)
+		$(INSTALL_DATA) $(libfactory) $(DESTDIR)$(libdir)/$(libfactory)
+		-$(INSTALL_DATA) libsingcf_g.a $(DESTDIR)$(libdir)/libsingcf_g.a
+		-$(INSTALL_DATA) libsingcf_p.a $(DESTDIR)$(libdir)/libsingcf_p.a
+		$(INSTALL_DATA) factory.h $(DESTDIR)$(includedir)/factory.h
+		$(INSTALL_DATA) cf_gmp.h $(DESTDIR)$(includedir)/cf_gmp.h
+		$(INSTALL_DATA) factoryconf.h $(DESTDIR)$(includedir)/factoryconf.h
 		for file in $(templateincl); do \
-		  $(INSTALL_DATA) $(srcdir)/templates/$$file $(templatedir)/$$file; \
+		  $(INSTALL_DATA) $(srcdir)/templates/$$file $(DESTDIR)$(templatedir)/$$file; \
 		done
 # we run `ranlib' another time for SUN machines
-		$(RANLIB) $(libdir)/$(libfactory)
+		$(RANLIB) $(DESTDIR)$(libdir)/$(libfactory)
 
 installmem:	mem
-		$(MKINSTALLDIRS) $(libdir)
-		$(INSTALL_DATA) libcfmem.a $(libdir)/libcfmem.a
+		$(MKINSTALLDIRS) $(DESTDIR)$(libdir)
+		$(INSTALL_DATA) libcfmem.a $(DESTDIR)$(libdir)/libcfmem.a
 # once again: another run for SUN
-		$(RANLIB) $(libdir)/libcfmem.a
+		$(RANLIB) $(DESTDIR)$(libdir)/libcfmem.a
 
 # to a pity, this cannot be done automatically ...
 installgftables:
 		@echo "Copying GF(q) tables ..."
 		@if test -d gftables; then \
-		  $(MKINSTALLDIRS) $(gftabledir); \
+		  $(MKINSTALLDIRS) $(DESTDIR)$(gftabledir); \
 		  for file in gftables/gftable.*.*; do \
-		    $(INSTALL_DATA) $$file $(gftabledir); \
+		    $(INSTALL_DATA) $$file $(DESTDIR)$(gftabledir); \
 		  done \
 		elif test -d "$(srcdir)"/gftables; then \
-		  $(MKINSTALLDIRS) $(gftabledir); \
+		  $(MKINSTALLDIRS) $(DESTDIR)$(gftabledir); \
 		  for file in "$(srcdir)"/gftables/gftable.*.*; do \
-		    $(INSTALL_DATA) $$file $(gftabledir); \
+		    $(INSTALL_DATA) $$file $(DESTDIR)$(gftabledir); \
 		  done \
 		else \
 		  echo "Run 'make gftables' before installing them"; \
diff -up Singular-3-1-6/IntegerProgramming/Makefile.in.destdir Singular-3-1-6/IntegerProgramming/Makefile.in
--- Singular-3-1-6/IntegerProgramming/Makefile.in.destdir	2012-12-19 15:01:16.000000000 -0600
+++ Singular-3-1-6/IntegerProgramming/Makefile.in	2014-03-10 09:48:46.354968360 -0500
@@ -63,12 +63,12 @@ install-nolns: install
 install_all: install
 
 install install-libsingular: $(MAIN1) $(MAIN2) $(MAIN3) $(MAIN4) $(LLL)
-	${MKINSTALLDIRS} ${bindir}
-	${INSTALL_PROGRAM} $(MAIN1) ${bindir}
-	${INSTALL_PROGRAM} $(MAIN2) ${bindir}
-	${INSTALL_PROGRAM} $(MAIN3) ${bindir}
-	${INSTALL_PROGRAM} $(MAIN4) ${bindir}
-	${INSTALL_PROGRAM} $(LLL) ${bindir}
+	${MKINSTALLDIRS} $(DESTDIR)${bindir}
+	${INSTALL_PROGRAM} $(MAIN1) $(DESTDIR)${bindir}
+	${INSTALL_PROGRAM} $(MAIN2) $(DESTDIR)${bindir}
+	${INSTALL_PROGRAM} $(MAIN3) $(DESTDIR)${bindir}
+	${INSTALL_PROGRAM} $(MAIN4) $(DESTDIR)${bindir}
+	${INSTALL_PROGRAM} $(LLL) $(DESTDIR)${bindir}
 
 install-bindist: $(MAIN1) $(MAIN2) $(MAIN3) $(MAIN4)
 	${MKINSTALLDIRS} ${install_bindir}
diff -up Singular-3-1-6/kernel/Makefile.in.destdir Singular-3-1-6/kernel/Makefile.in
--- Singular-3-1-6/kernel/Makefile.in.destdir	2012-12-19 15:01:16.000000000 -0600
+++ Singular-3-1-6/kernel/Makefile.in	2014-03-10 09:48:46.354968360 -0500
@@ -272,9 +272,9 @@ p_Procs_%.so: p_Procs_Lib_%.dl_o
 install-nolns: install
 
 install-libsingular: install
-	-${MKINSTALLDIRS} ${includedir}
-	-${MKINSTALLDIRS} ${includedir}/singular
-	-for file in *.h kInline.cc; do sed -e "s:<kernel/:<singular/:"< $$file | sed -e "s:<Singular/:<singular/:"|sed -e "s:<omalloc/:<:"|sed -e "s:<factory/:<:" > ${includedir}/singular/$$file; done
+	-${MKINSTALLDIRS} $(DESTDIR)${includedir}
+	-${MKINSTALLDIRS} $(DESTDIR)${includedir}/singular
+	-for file in *.h kInline.cc; do sed -e "s:<kernel/:<singular/:"< $$file | sed -e "s:<Singular/:<singular/:"|sed -e "s:<omalloc/:<:"|sed -e "s:<factory/:<:" > $(DESTDIR)${includedir}/singular/$$file; done
 
 install: all installbin
 
@@ -290,12 +290,12 @@ endif
 
 
 installbin: ${DL_LIBS} ${DL_LIBSG} libkernel.a libkernel_g.a
-	${MKINSTALLDIRS} ${bindir}
+	${MKINSTALLDIRS} $(DESTDIR)${bindir}
 	- ranlib libkernel.a
 	- ranlib libkernel_g.a
 ifeq (,${DL_LIBS})
 else
-	-${INSTALL_PROGRAM} ${DL_LIBS} ${bindir}
+	-${INSTALL_PROGRAM} ${DL_LIBS} $(DESTDIR)${bindir}
 ifeq (,${DL_LIBSG})
 else
 	-for file in ${DL_LIBS} ${DL_LIBSG}; do ${INSTALL_PROGRAM} $$file ../Singular; done
@@ -304,10 +304,10 @@ endif
 
 
 install-bindist: ${DL_LIBS} libkernel.a
-	${MKINSTALLDIRS} ${bindir}
+	${MKINSTALLDIRS} $(DESTDIR)${bindir}
 ifeq (,${DL_LIBS})
 else
-	-${INSTALL_PROGRAM} ${DL_LIBS} ${bindir}
+	-${INSTALL_PROGRAM} ${DL_LIBS} $(DESTDIR)${bindir}
 endif
 
 ##
diff -up Singular-3-1-6/libfac/Makefile.in.destdir Singular-3-1-6/libfac/Makefile.in
--- Singular-3-1-6/libfac/Makefile.in.destdir	2012-12-19 15:01:16.000000000 -0600
+++ Singular-3-1-6/libfac/Makefile.in	2014-03-10 09:48:46.354968360 -0500
@@ -146,17 +146,17 @@ libsingfac_p.a:	$(PROFOBJ)
 install-nolns: install
 
 install install-libsingular:	all
-	${MKINSTALLDIRS} ${includedir}
-	${MKINSTALLDIRS} ${libdir}
-	${INSTALL_DATA} ${LIBFACINCLUDE} ${includedir}
-	${INSTALL_DATA} ${TARGETNAME}.a ${libdir}
-	-${INSTALL_DATA} libsingfac_g.a ${libdir}
-	$(RANLIB) ${libdir}/${TARGETNAME}.a
-	-$(RANLIB) ${libdir}/libsingfac_g.a
+	${MKINSTALLDIRS} $(DESTDIR)${includedir}
+	${MKINSTALLDIRS} $(DESTDIR)${libdir}
+	${INSTALL_DATA} ${LIBFACINCLUDE} $(DESTDIR)${includedir}
+	${INSTALL_DATA} ${TARGETNAME}.a $(DESTDIR)${libdir}
+	-${INSTALL_DATA} libsingfac_g.a $(DESTDIR)${libdir}
+	$(RANLIB) $(DESTDIR)${libdir}/${TARGETNAME}.a
+	-$(RANLIB) $(DESTDIR)${libdir}/libsingfac_g.a
 
 install_all: install libsingfac_p.a
-	-${INSTALL_DATA} libsingfac_p.a ${libdir}
-	-$(RANLIB) ${libdir}/libsingfac_p.a
+	-${INSTALL_DATA} libsingfac_p.a $(DESTDIR)${libdir}
+	-$(RANLIB) $(DESTDIR)${libdir}/libsingfac_p.a
 
 uninstall: 
 	$(RM) ${includedir}/${LIBFACINCLUDE}
diff -up Singular-3-1-6/omalloc/Makefile.in.destdir Singular-3-1-6/omalloc/Makefile.in
--- Singular-3-1-6/omalloc/Makefile.in.destdir	2012-12-19 15:01:16.000000000 -0600
+++ Singular-3-1-6/omalloc/Makefile.in	2014-03-10 09:48:46.355968349 -0500
@@ -181,35 +181,35 @@ configure: configure.in
 install-nolns: install
 
 install-libsingular:
-	$(MKINSTALLDIRS) $(includedir)/singular
-	$(INSTALL) omalloc.h $(includedir)/singular
+	$(MKINSTALLDIRS) $(DESTDIR)$(includedir)/singular
+	$(INSTALL) omalloc.h $(DESTDIR)$(includedir)/singular
 
 install install-libsingular: all
-	$(MKINSTALLDIRS) $(libdir)
-	$(MKINSTALLDIRS) $(includedir)
-	$(INSTALL_DATA) libomalloc.a $(libdir)
-	$(RANLIB) $(libdir)/libomalloc.a
-	$(INSTALL_DATA) libomalloc_ndebug.a $(libdir)
-	$(RANLIB) $(libdir)/libomalloc_ndebug.a
-	$(INSTALL_DATA) omalloc.o $(libdir)
-	$(INSTALL_DATA) omalloc_debug.o $(libdir)
-	$(INSTALL_DATA) omalloc.h $(includedir)
-	$(INSTALL_DATA) omlimits.h $(includedir)
-	$(INSTALL_DATA) mylimits.h $(includedir)
+	$(MKINSTALLDIRS) $(DESTDIR)$(libdir)
+	$(MKINSTALLDIRS) $(DESTDIR)$(includedir)
+	$(INSTALL_DATA) libomalloc.a $(DESTDIR)$(libdir)
+	$(RANLIB) $(DESTDIR)$(libdir)/libomalloc.a
+	$(INSTALL_DATA) libomalloc_ndebug.a $(DESTDIR)$(libdir)
+	$(RANLIB) $(DESTDIR)$(libdir)/libomalloc_ndebug.a
+	$(INSTALL_DATA) omalloc.o $(DESTDIR)$(libdir)
+	$(INSTALL_DATA) omalloc_debug.o $(DESTDIR)$(libdir)
+	$(INSTALL_DATA) omalloc.h $(DESTDIR)$(includedir)
+	$(INSTALL_DATA) omlimits.h $(DESTDIR)$(includedir)
+	$(INSTALL_DATA) mylimits.h $(DESTDIR)$(includedir)
 
 install_all: install libomalloc_p.a
-	$(INSTALL_DATA) libomalloc_p.a $(libdir)
-	$(RANLIB) $(libdir)/libomalloc_p.a
+	$(INSTALL_DATA) libomalloc_p.a $(DESTDIR)$(libdir)
+	$(RANLIB) $(DESTDIR)$(libdir)/libomalloc_p.a
 
 install_min: libomalloc_ndebug.a omalloc.o
-	$(MKINSTALLDIRS) $(libdir)
-	$(MKINSTALLDIRS) $(includedir)
-	$(INSTALL_DATA) libomalloc_ndebug.a $(libdir)
-	$(RANLIB) $(libdir)/libomalloc_ndebug.a
-	$(INSTALL_DATA) omalloc.o $(libdir)
-	$(INSTALL_DATA) omalloc.h $(includedir)
-	$(INSTALL_DATA) omlimits.h $(includedir)
-	$(INSTALL_DATA) mylimits.h $(includedir)
+	$(MKINSTALLDIRS) $(DESTDIR)$(libdir)
+	$(MKINSTALLDIRS) $(DESTDIR)$(includedir)
+	$(INSTALL_DATA) libomalloc_ndebug.a $(DESTDIR)$(libdir)
+	$(RANLIB) $(DESTDIR)$(libdir)/libomalloc_ndebug.a
+	$(INSTALL_DATA) omalloc.o $(DESTDIR)$(libdir)
+	$(INSTALL_DATA) omalloc.h $(DESTDIR)$(includedir)
+	$(INSTALL_DATA) omlimits.h $(DESTDIR)$(includedir)
+	$(INSTALL_DATA) mylimits.h $(DESTDIR)$(includedir)
 
 uninstall:
 	rm -f $(includedir)/omalloc.h
diff -up Singular-3-1-6/Singular/Makefile.in.destdir Singular-3-1-6/Singular/Makefile.in
--- Singular-3-1-6/Singular/Makefile.in.destdir	2012-12-19 15:01:16.000000000 -0600
+++ Singular-3-1-6/Singular/Makefile.in	2014-03-10 09:49:43.986351393 -0500
@@ -562,57 +562,58 @@ install_all: install
 install-nolns: all installbin-nolns installslib-nolns
 
 installbin: ${SING_EXEC} libparse ESingular TSingular ${RUN_SINGULARS}
-	${MKINSTALLDIRS} ${bindir} ${bindir_mod}
-	${INSTALL_PROGRAM} ${SING_EXEC} ${SINGULAR}
+	${MKINSTALLDIRS} $(DESTDIR)${bindir} $(DESTDIR)${bindir_mod}
+	${INSTALL_PROGRAM} ${SING_EXEC} $(DESTDIR)${SINGULAR}
 	-for file in *.so; do \
-	  ${INSTALL_PROGRAM} $$file ${bindir}; \
+	  ${INSTALL_PROGRAM} $$file $(DESTDIR)${bindir}; \
 	done
 	-for file in MOD/*.so; do \
-	  ${INSTALL_PROGRAM} $$file ${bindir}/MOD; \
+	  ${INSTALL_PROGRAM} $$file $(DESTDIR)${bindir}/MOD; \
 	done
-	${INSTALL_PROGRAM} libparse ${RUN_SINGULARS} ${bindir}
-	${INSTALL_PROGRAM} ESingular TSingular ${bindir}
-	${INSTALL_PROGRAM} surfex ${bindir}
-	chmod a+x ${SINGULAR}
-	rm -f ${bindir}/${SING_EXEC}${EXEC_EXT}
-	cd ${bindir}; ${LN_S} ${SINGULAR} Singular${EXEC_EXT};
+	${INSTALL_PROGRAM} libparse ${RUN_SINGULARS} $(DESTDIR)${bindir}
+	${INSTALL_PROGRAM} ESingular TSingular $(DESTDIR)${bindir}
+	${INSTALL_PROGRAM} surfex $(DESTDIR)${bindir}
+	chmod a+x $(DESTDIR)${SINGULAR}
+	rm -f $(DESTDIR)${bindir}/${SING_EXEC}${EXEC_EXT}
+	cd $(DESTDIR)${bindir}; ${LN_S} ${SINGULAR} Singular${EXEC_EXT};
 
 installbin-nolns: ${SING_EXEC} libparse ESingular TSingular ${RUN_SINGULARS}
-	${MKINSTALLDIRS} ${bindir} ${bindir_mod}
-	${INSTALL_PROGRAM} ${SING_EXEC} ${SINGULAR}
+	${MKINSTALLDIRS} $(DESTDIR)${bindir} $(DESTDIR)${bindir_mod}
+	${INSTALL_PROGRAM} ${SING_EXEC} $(DESTDIR)${SINGULAR}
 	-for file in *.so; do \
-	  ${INSTALL_PROGRAM} $$file ${bindir}; \
+	  ${INSTALL_PROGRAM} $$file $(DESTDIR)${bindir}; \
 	done
 	-for file in MOD/*.so; do \
-	  ${INSTALL_PROGRAM} $$file ${bindir}/MOD; \
+	  ${INSTALL_PROGRAM} $$file $(DESTDIR)${bindir}/MOD; \
 	done
-	${INSTALL_PROGRAM} libparse ${RUN_SINGULARS} ${bindir}
-	${INSTALL_PROGRAM} ESingular TSingular ${DL_LIBS} ${bindir}
-	chmod a+x ${SINGULAR}
-	rm -f ${bindir}/${SING_EXEC}${EXEC_EXT}
-	${INSTALL_PROGRAM} ${SINGULAR} ${bindir}/Singular${EXEC_EXT};
+	${INSTALL_PROGRAM} libparse ${RUN_SINGULARS} $(DESTDIR)${bindir}
+	${INSTALL_PROGRAM} ESingular TSingular ${DL_LIBS} $(DESTDIR)${bindir}
+	chmod a+x $(DESTDIR)${SINGULAR}
+	rm -f $(DESTDIR)${bindir}/${SING_EXEC}${EXEC_EXT}
+	${INSTALL_PROGRAM} ${SINGULAR} $(DESTDIR)${bindir}/Singular${EXEC_EXT};
 
 installslib: LIB
-	test -r ${slibdir} || ${LN_S} `pwd`/LIB ${slibdir}
+	#test -r ${slibdir} || ${LN_S} `pwd`/LIB ${slibdir}
+	:
 
 installslib-nolns: LIB
-	${MKINSTALLDIRS} ${slibdir}
+	${MKINSTALLDIRS} $(DESTDIR)${slibdir}
 	for file in `pwd`/LIB/*.lib; do \
-	  ${INSTALL_DATA}  $$file ${slibdir}; \
+	  ${INSTALL_DATA}  $$file $(DESTDIR)${slibdir}; \
 	done
-	${MKINSTALLDIRS} ${slibdir}/gftables
+	${MKINSTALLDIRS} $(DESTDIR)${slibdir}/gftables
 	for file in `pwd`/LIB/gftables/*; do \
-	  ${INSTALL_DATA}  $$file ${slibdir}/gftables/; \
+	  ${INSTALL_DATA}  $$file $(DESTDIR)${slibdir}/gftables/; \
 	done
 
 install-libsingular: libsingular installslib-nolns
-	${MKINSTALLDIRS} ${includedir}/singular
+	${MKINSTALLDIRS} $(DESTDIR)${includedir}/singular
 	for file in *.$(SO_SUFFIX); do \
-	  ${INSTALL_PROGRAM}  $$file ${libdir}; \
+	  ${INSTALL_PROGRAM}  $$file $(DESTDIR)${libdir}; \
 	done
-	${INSTALL_PROGRAM} libsingular.h ${includedir}
-	for file in subexpr.h tok.h grammar.h ipid.h lists.h ipshell.h attrib.h silink.h table.h; do \
-	sed -e "s:<kernel/:<singular/:" < $$file |sed -e "s:<Singular/:<singular/:"|sed -e "s:<omalloc/:<:"|sed -e "s:<factory/:<:" >${includedir}/singular/$$file ;\
+	${INSTALL_PROGRAM} libsingular.h $(DESTDIR)${includedir}
+	for file in subexpr.h tok.h grammar.h ipid.h lists.h ipshell.h attrib.h silink.h sing_dbm.h table.h; do \
+	sed -e "s:<kernel/:<singular/:" < $$file |sed -e "s:<Singular/:<singular/:"|sed -e "s:<omalloc/:<:"|sed -e "s:<factory/:<:" >$(DESTDIR)${includedir}/singular/$$file ;\
 	done
 
 uninstall: uninstallbin
