--- singular-4.1.0/kernel/linear_algebra/CacheImplementation.h.orig	2017-02-20 10:33:27.000000000 -0700
+++ singular-4.1.0/kernel/linear_algebra/CacheImplementation.h	2017-02-28 08:35:19.070578172 -0700
@@ -354,7 +354,7 @@ bool Cache<KeyClass, ValueClass>::put (c
 template<class KeyClass, class ValueClass>
 std::string Cache<KeyClass, ValueClass>::toString() const
 {
-  char h[10];
+  char h[12];
   std::string s = "Cache:";
   s += "\n   entries: ";
   sprintf(h, "%d", getNumberOfEntries()); s += h;
--- singular-4.1.0/Singular/extra.cc.orig	2017-02-20 10:33:27.000000000 -0700
+++ singular-4.1.0/Singular/extra.cc	2017-02-28 08:36:17.566405458 -0700
@@ -2499,7 +2499,7 @@ static BOOLEAN jjEXTENDED_SYSTEM(leftv r
         if(strcmp(sys_cmd,"changeRing")==0)
         {
           int varN = currRing->N;
-          char h[10];
+          char h[12];
           for (int i = 1; i <= varN; i++)
           {
             omFree(currRing->names[i - 1]);
