--- singular-4.1.0/emacs/.emacs-singular.orig	2017-02-20 10:33:27.000000000 -0700
+++ singular-4.1.0/emacs/.emacs-singular	2017-04-20 21:15:06.267460576 -0600
@@ -185,7 +185,7 @@
 		  "" "/")))
 
 ;; add global menus for Singular, Emacs version
-(unless (fboundp 'add-submenu)
+(unless (boundp 'current-menubar)
   (setq singular-start-map (make-sparse-keymap))
   (define-key singular-start-map [menu-bar singular]
     (cons "Singular" (make-sparse-keymap "Singular")))
@@ -197,7 +197,7 @@
   (use-local-map singular-start-map))
 
 ;; add global menus for Singular, XEmacs version
-(when (fboundp 'add-submenu)
+(when (boundp 'current-menubar)
   (add-submenu nil
 	       '("Singular"
 		 ["Start default" singular t]
--- singular-4.1.0/resources/feResource.cc.orig	2017-02-20 10:33:27.000000000 -0700
+++ singular-4.1.0/resources/feResource.cc	2017-04-19 20:10:05.304202866 -0600
@@ -83,7 +83,7 @@ feResourceConfig_s feResourceConfigs[] =
   {"RootDir",   'r',    feResDir,   "SINGULAR_ROOT_DIR",    "%b/..",                (char *)""},
   {"DataDir",   'D',    feResDir,   "SINGULAR_DATA_DIR",    "%b/../share/",          (char *)""},
   {"DefaultDir",'d',    feResDir,   "SINGULAR_DEFAULT_DIR",  SINGULAR_DEFAULT_DIR,  (char *)""},
-  {"InfoFile",  'i',    feResFile,  "SINGULAR_INFO_FILE",   "%D/info/singular.hlp", (char *)""},
+  {"InfoFile",  'i',    feResFile,  "SINGULAR_INFO_FILE",   "%D/info/singular.hlp.gz", (char *)""},
   {"IdxFile",   'x',    feResFile,  "SINGULAR_IDX_FILE",    "%D/singular/singular.idx",  (char *)""},
   {"HtmlDir",   'h',    feResDir,   "SINGULAR_HTML_DIR",    "%D/singular/html",              (char *)""},
   {"ManualUrl", 'u',    feResUrl,   "SINGULAR_URL",         "http://www.singular.uni-kl.de/Manual/",    (char *)""},
