diff -up Singular-3-1-7/kernel/Makefile.in.orig Singular-3-1-7/kernel/Makefile.in
--- Singular-3-1-7/kernel/Makefile.in.orig	2016-02-15 20:50:21.528455435 -0700
+++ Singular-3-1-7/kernel/Makefile.in	2016-02-15 21:06:08.429066185 -0700
@@ -48,7 +48,7 @@ PIPE		= @PIPE@
 CFLAGS		= @CFLAGS@ ${PIPE} 
 CXXFLAGS	= @CXXFLAGS@ ${PIPE} 
 CXXTEMPLFLAGS	= @CXXTEMPLFLAGS@
-CPPFLAGS	= -I${srcdir} -I.. -I@prefix@  @CPPFLAGS@ 
+CPPFLAGS	= -I${srcdir} -I.. -I../omalloc -I${srcdir}/../factory -I@prefix@ @CPPFLAGS@
 DEFS		= -DNDEBUG -DOM_NDEBUG -D@SING_UNAME@ @DEFS@
 LDFLAGS		= @LDFLAGS@
 LD_DYN_FLAGS	= @LD_DYN_FLAGS@
diff -up Singular-3-1-7/libfac/Makefile.in.orig Singular-3-1-7/libfac/Makefile.in
--- Singular-3-1-7/libfac/Makefile.in.orig	2016-02-15 20:50:21.528455435 -0700
+++ Singular-3-1-7/libfac/Makefile.in	2016-02-15 21:07:06.131411169 -0700
@@ -36,7 +36,7 @@ MKINSTALLDIRS	= ./mkinstalldirs
 ##
 CXXFLAGS	= @CXXFLAGS@
 CXXTEMPLFLAGS	= @CXXTEMPLFLAGS@
-CPPFLAGS	= -I${srcdir}/factor -I${srcdir}/charset -I${srcdir} -I.. -I${prefix} -I${prefix}/include @CPPFLAGS@ 
+CPPFLAGS	= -I${srcdir}/factor -I${srcdir}/charset -I${srcdir} -I${srcdir}/.. -I${srcdir}/../factory -I${prefix} -I${prefix}/include @CPPFLAGS@
 DEFS		= @DEFS@ 
 LDFLAGS		= @LDFLAGS@
 LIBS		= -lfac -lcf -lcfmem -lgmp @LIBS@
diff -up Singular-3-1-7/Singular/Makefile.in.orig Singular-3-1-7/Singular/Makefile.in
--- Singular-3-1-7/Singular/Makefile.in.orig	2016-02-15 20:58:01.359359527 -0700
+++ Singular-3-1-7/Singular/Makefile.in	2016-02-15 21:07:59.903073250 -0700
@@ -87,7 +87,7 @@ CFLAGS		= @CFLAGS@ ${PIPE}
 CXXFLAGS	= @CXXFLAGS@ -I.. -I@prefix@ ${PIPE}
 CXXTEMPLFLAGS	= @CXXTEMPLFLAGS@ -I.. -I@prefix@
 ifneq ($(SINGUNAME),ix86-Win)
-CPPFLAGS	= -I${srcdir} -I.. -I@prefix@ @CPPFLAGS@ @PYTHON_CPPFLAGS@ @FLINT_CFLAGS@
+CPPFLAGS	= -I${srcdir} -I${srcdir}/.. -I${srcdir}/../omalloc -I@prefix@ @CPPFLAGS@ @PYTHON_CPPFLAGS@ @FLINT_CFLAGS@
 else
 ## Dirs after @CPPFLAGS@ are for PySingular
 CPPFLAGS	= -I${srcdir} -I.. -I@prefix@ @CPPFLAGS@ @PYTHON_CPPFLAGS@ @FLINT_CFLAGS@
