--- singular-4.1.0/gfanlib/gfanlib_tropicalhomotopy.h.orig	2017-02-20 10:33:27.000000000 -0700
+++ singular-4.1.0/gfanlib/gfanlib_tropicalhomotopy.h	2017-02-28 08:51:27.144697675 -0700
@@ -454,7 +454,7 @@ template<class mvtyp, class mvtypDouble,
                         //chioices are "relative" so no update is needed.
 
                         choices=parent.choices;
-                        int numberToDrop=(subconfigurationIndex!=0) ? numberToDrop=k+1 : 0;
+                        int numberToDrop=(subconfigurationIndex!=0) ? k+1 : 0;
 
                         choices[subconfigurationIndex-1].first-=numberToDrop;
                         choices[subconfigurationIndex-1].second-=numberToDrop;
