diff -up Singular-3-1-5/Singular/semaphore.c.orig Singular-3-1-5/Singular/semaphore.c
--- Singular-3-1-5/Singular/semaphore.c.orig	2012-12-19 14:01:16.000000000 -0700
+++ Singular-3-1-5/Singular/semaphore.c	2013-05-02 08:53:05.707047549 -0600
@@ -4,6 +4,8 @@
 #include <sys/stat.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
 #include <Singular/mod2.h>
 
 #ifdef HAVE_SIMPLEIPC
