diff -up Singular-3-1-7/Singular/semaphore.c.orig Singular-3-1-7/Singular/semaphore.c
--- Singular-3-1-7/Singular/semaphore.c.orig	2014-08-06 09:59:15.000000000 -0600
+++ Singular-3-1-7/Singular/semaphore.c	2016-02-15 21:36:08.138939362 -0700
@@ -4,6 +4,8 @@
 #include <sys/stat.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
 #include <Singular/mod2.h>
 #include <Singular/cntrlc.h>
 #include <Singular/si_signals.h>
