diff -up Singular-3-1-5/factory/FLINTconvert.cc.orig Singular-3-1-5/factory/FLINTConvert.cc
--- Singular-3-1-5/factory/FLINTconvert.cc.orig	2012-07-12 08:24:57.000000000 -0600
+++ Singular-3-1-5/factory/FLINTconvert.cc	2014-03-06 16:34:20.594876371 -0700
@@ -269,9 +269,9 @@ convertFLINTnmod_poly_factor2FacCFFList
 
   long i;
 
-  for (i = 0; i < fac->num_factors; i++)
-    result.append (CFFactor (convertnmod_poly_t2FacCF ((nmod_poly_t &)fac->factors[i],x),
-                             fac->exponents[i]));
+  for (i = 0; i < fac->num; i++)
+    result.append (CFFactor (convertnmod_poly_t2FacCF (&fac->p[i],x),
+                             fac->exp[i]));
   return result;
 }
 
diff -up Singular-3-1-5/Singular/Makefile.in.flint24 Singular-3-1-5/Singular/Makefile.in
--- Singular-3-1-5/Singular/Makefile.in.flint24	2014-03-10 13:48:54.812129117 -0500
+++ Singular-3-1-5/Singular/Makefile.in	2014-03-10 13:50:12.069307410 -0500
@@ -132,7 +132,7 @@ HAVE_MPSR       = @HAVE_MPSR@
 #
 # Handle libSINGULAR stuff
 #
-LIBSINGULAR_LIBS =-lsingfac -lsingcf -lntl -lreadline -lgmp -lomalloc -lpthread -ldl
+LIBSINGULAR_LIBS =-lflint -lsingfac -lsingcf -lntl -lreadline -lgmp -lomalloc -lpthread -ldl
 
 SO_SUFFIX = so
 
