--- Singular-3-1-5/factory/NTLconvert.cc.orig	2012-12-19 14:01:16.000000000 -0700
+++ Singular-3-1-5/factory/NTLconvert.cc	2013-05-03 08:36:16.564528847 -0600
@@ -55,6 +55,27 @@ int fac_NTL_char=-1;            // the c
 NTL_CLIENT
 #endif
 
+// Needed explicit template instantiations for NTL 6.x.x
+template mat_zz_p::Mat(const mat_zz_p&);
+template mat_zz_pE::Mat(const mat_zz_pE&);
+
+template vec_pair_GF2X_long&
+vec_pair_GF2X_long::operator=(const vec_pair_GF2X_long&);
+template vec_pair_GF2EX_long&
+vec_pair_GF2EX_long::operator=(const vec_pair_GF2EX_long&);
+template vec_pair_zz_pX_long&
+vec_pair_zz_pX_long::operator=(const vec_pair_zz_pX_long&);
+template vec_pair_zz_pEX_long&
+vec_pair_zz_pEX_long::operator=(const vec_pair_zz_pEX_long&);
+template vec_pair_ZZ_pX_long&
+vec_pair_ZZ_pX_long::operator=(const vec_pair_ZZ_pX_long&);
+template vec_pair_ZZX_long&
+vec_pair_ZZX_long::operator=(const vec_pair_ZZX_long&);
+template vec_vec_zz_p& vec_vec_zz_p::operator=(const vec_vec_zz_p&);
+template vec_vec_zz_pE& vec_vec_zz_pE::operator=(const vec_vec_zz_pE&);
+
+template long operator==<zz_p>(const vec_zz_p&, const vec_zz_p&);
+
 ////////////////////////////////////////////////////////////////////////////////
 // NAME: convertFacCF2NTLZZpX                                                 //
 //                                                                            //
