diff -up Singular-3-1-5/doc/Makefile.in.orig Singular-3-1-5/doc/Makefile.in
--- Singular-3-1-5/doc/Makefile.in.orig	2012-08-04 17:58:04.824209703 -0400
+++ Singular-3-1-5/doc/Makefile.in	2012-08-04 17:59:02.478211895 -0400
@@ -62,14 +62,14 @@ SINGULAR_LIB_DIR= ../Singular/LIB
 ifeq ($(SINGUNAME),"ix86-Win")
 LIBPARSE        = ${bindir}/libparse.exe
 else
-LIBPARSE        = ${bindir}/libparse
+LIBPARSE        = $(DESTDIR)${bindir}/libparse
 endif
 DOC_SUBDIR      = ./d2t_singular
 EX_SUBDIR       = ./examples
 CHKSUM_DB       = ${DOC_SUBDIR}/chksum
 DOC2TEX_EXAMPLE_EXCLUSIONS = @DOC2TEX_EXAMPLE_EXCLUSIONS@
 DOC2TEX         = ${PERL} ./doc2tex.pl -docdir ${DOC_SUBDIR} \
-                  -Singular $(DESTDIR)${SINGULAR} -verbose ${VERBOSE} -make ${MAKE} \
+                  -Singular /bin/true -verbose ${VERBOSE} -make ${MAKE} \
 		  -exdir ${EX_SUBDIR} ${DOC2TEX_EXAMPLE_EXCLUSIONS}
 TAG		=
 PL2DOC		= ${PERL} ./pl2doc.pl ${TAG} -db ${CHKSUM_DB}
@@ -322,7 +322,6 @@ install-sharedist: singular.hlp html sin
 	${MKINSTALLDIRS} ${install_htmldir}
 	cp ${HTML_SUBDIR}/*.htm ${install_htmldir}
 	cp ${HTML_SUBDIR}/*.gif ${install_htmldir}
-	-cp ${HTML_SUBDIR}/*.png ${install_htmldir}
 	cp ${HTML_SUBDIR}/*.jpg ${install_htmldir}
 	chmod -R +rX ${install_htmldir}
 	${MKINSTALLDIRS} ${install_docdir}
diff -up Singular-3-1-5/emacs/Makefile.in.orig Singular-3-1-5/emacs/Makefile.in
--- Singular-3-1-5/emacs/Makefile.in.orig	2012-08-04 17:59:10.208212218 -0400
+++ Singular-3-1-5/emacs/Makefile.in	2012-08-04 18:01:33.510217756 -0400
@@ -6,11 +6,12 @@
 SHELL=/bin/sh
 
 prefix		= @prefix@
-docdir		= ${prefix}/doc
-examplesdir	= ${prefix}/examples
+libdir		= @libdir@
+datadir		= @datadir@
 SINGULAR	= @SINGULAR@
-singulardir	= ${prefix}/Singular 
-emacsdir	= ${install_prefix}/emacs
+singulardir	= ${libdir}/Singular
+examplesdir	= ${singulardir}/examples
+emacsdir	= ${datadir}/emacs/site-lisp/singular
 
 
 PERL		= @PERL@
@@ -18,42 +19,42 @@ INSTALL		= ../install-sh -c
 INSTALL_DATA	= ${INSTALL} -m 644
 MKINSTALLDIRS   = ../mkinstalldirs
 
-SINGULAR_HLP	= ${docdir}/singular.hlp
+SINGULAR_HLP	= ${singulardir}/info/singular.hlp
 
 all install: cmd-cmpl.el hlp-cmpl.el lib-cmpl.el ex-cmpl.el
 
 install_all: install
 
-cmd-cmpl.el: ${SINGULAR} cmpl.pl
-	${PERL} cmpl.pl -Singular ${SINGULAR} -cmpl cmd > cmd-cmpl.el
+cmd-cmpl.el: cmpl.pl
+	${PERL} cmpl.pl -Singular $(DESTDIR)${SINGULAR} -cmpl cmd > cmd-cmpl.el
 
-hlp-cmpl.el: ${SINGULAR_HLP} cmpl.pl
-	${PERL} cmpl.pl -hlp ${SINGULAR_HLP} -cmpl hlp > hlp-cmpl.el
+hlp-cmpl.el: cmpl.pl
+	${PERL} cmpl.pl -hlp $(DESTDIR)${SINGULAR_HLP} -cmpl hlp > hlp-cmpl.el
 
-ex-cmpl.el: cmpl.pl ${examplesdir}
-		${PERL} cmpl.pl -cmpl ex -ex_dir ${examplesdir} > ex-cmpl.el
+ex-cmpl.el: cmpl.pl
+	${PERL} cmpl.pl -cmpl ex -ex_dir $(DESTDIR)${examplesdir} > ex-cmpl.el
 
-lib-cmpl-old.el: ${prefix}/Singular/Makefile cmpl.pl
-	${MAKE} --no-print-directory -s -f ${prefix}/Singular/Makefile echoLIBS > lib-temp
+lib-cmpl-old.el: cmpl.pl
+	${MAKE} --no-print-directory -s -f ../Singular/Makefile echoLIBS > lib-temp
 	cat lib-temp | ${PERL} cmpl.pl -cmpl lib > lib-cmpl-old.el
 	rm lib-temp
 
-lib-cmpl.el: ${prefix}/Singular/Makefile lib.pl
-	(cd ${prefix}/Singular;${MAKE} --no-print-directory -s -f Makefile echoLIBS) > lib-temp
-	${PERL} lib.pl -s ${singulardir} `cat lib-temp` > lib-cmpl.el
+lib-cmpl.el: lib.pl
+	(cd ../Singular;${MAKE} --no-print-directory -s -f Makefile echoLIBS) > lib-temp
+	${PERL} lib.pl -s $(DESTDIR)${singulardir} `cat lib-temp` > lib-cmpl.el
 
 clean mostlyclean distclean:
 	rm -f cmd-cmpl.el hlp-cmpl.el lib-cmpl.el ex-cmpl.el
 
 install-sharedist: all
-	${MKINSTALLDIRS} ${emacsdir}
-	${INSTALL_DATA} cmd-cmpl.el ${emacsdir}
-	${INSTALL_DATA} lib-cmpl.el ${emacsdir}
-	${INSTALL_DATA} hlp-cmpl.el ${emacsdir}
-	${INSTALL_DATA} ex-cmpl.el ${emacsdir}
-	${INSTALL_DATA} singular.el ${emacsdir}
-	${INSTALL_DATA} .emacs-singular ${emacsdir}
-	${INSTALL_DATA} .emacs-general ${emacsdir}
+	${MKINSTALLDIRS} $(DESTDIR)${emacsdir}
+	${INSTALL_DATA} cmd-cmpl.el $(DESTDIR)${emacsdir}
+	${INSTALL_DATA} lib-cmpl.el $(DESTDIR)${emacsdir}
+	${INSTALL_DATA} hlp-cmpl.el $(DESTDIR)${emacsdir}
+	${INSTALL_DATA} ex-cmpl.el $(DESTDIR)${emacsdir}
+	${INSTALL_DATA} singular.el $(DESTDIR)${emacsdir}
+	${INSTALL_DATA} .emacs-singular $(DESTDIR)${emacsdir}
+	${INSTALL_DATA} .emacs-general $(DESTDIR)${emacsdir}
 
 depend:
-	@echo "No make depend available!"
\ No newline at end of file
+	@echo "No make depend available!"
