--- singular-4.2.0/Singular/dyn_modules/gfanlib/Makefile.am.orig	2021-05-25 19:56:34.726983587 -0600
+++ singular-4.2.0/Singular/dyn_modules/gfanlib/Makefile.am	2021-05-25 19:56:56.827919382 -0600
@@ -41,5 +41,5 @@ endif
 
 if HAVE_GFANLIB
 # Use objects from the libtool convenience library
- gfanlib_la_LIBADD   += ${abs_top_builddir}/gfanlib/libgfan.la ${CDDGMPLDFLAGS}
+ gfanlib_la_LIBADD   += -lgfan
 endif
