diff -up Singular-3-1-6/kernel/Makefile.in.builddid Singular-3-1-6/kernel/Makefile.in
--- Singular-3-1-6/kernel/Makefile.in.builddid	2014-01-17 08:42:37.564086425 -0600
+++ Singular-3-1-6/kernel/Makefile.in	2014-01-17 08:42:37.574086316 -0600
@@ -264,7 +264,7 @@ p_Procs_Dynamic.inc: p_Procs_Generate.cc
 	-rm -f p_Procs_Dynamic p_Procs_Dynamic.exe
 
 p_Procs_%.so: p_Procs_Lib_%.dl_o
-	$(LD) ${SLDFLAGS} -o $@ $^
+	$(CXX) ${SLDFLAGS} -o $@ $^
 
 ##
 ## install targets
@@ -374,7 +374,7 @@ DEFSG		= -D@SING_UNAME@ @DEFS@ -DDL_TAIL
 ##
 
 p_Procs_%.sog: p_Procs_Lib_%.dl_og
-	$(LD) ${SLDFLAGS} -o $@ $^
+	$(CXX) ${SLDFLAGS} -o $@ $^
 
 p_Procs_Lib_%.dl_og : p_Procs_Lib.cc  p_Procs_Dynamic.og
 	${CXXG} ${CXXFLAGSG} ${CXXTEMPLFLAGS} ${CPPFLAGS} ${DEFSG} ${SFLAGS} -Dp_Procs_$* -c $< -o $@
diff -up Singular-3-1-6/Singular/Makefile.in.builddid Singular-3-1-6/Singular/Makefile.in
--- Singular-3-1-6/Singular/Makefile.in.builddid	2014-01-17 08:42:37.574086316 -0600
+++ Singular-3-1-6/Singular/Makefile.in	2014-01-17 08:43:25.953560965 -0600
@@ -489,10 +489,10 @@ feOptTS.inc: feOpt.cc mod2.h
 	rm -f feOptTS feOptTS.exe
 
 mpsr.so: $(MPSR_SOURCES:.cc=.dl_o)
-	$(LD) ${SLDFLAGS} -o $@ $^ -L${libdir} ${MP_LIBS} ${GLIBC_DYN_FIX}
+	$(LIBSINGULAR_LD) ${SLDFLAGS} -o $@ $^ -L${libdir} ${MP_LIBS} ${GLIBC_DYN_FIX}
 
 dbmsr.so: $(DBMSR_SOURCES:.cc=.dl_o)
-	$(LD) ${SLDFLAGS} -o $@ $^ ${GLIBC_DYN_FIX}
+	$(LIBSINGULAR_LD) ${SLDFLAGS} -o $@ $^ ${GLIBC_DYN_FIX}
 
 # dynamic module for python bject support
 MOD/pyobject.${MODULE_SUFFIX}: pyobject.dl_o
