--- singular-4.1.1/configure.ac.orig	2018-07-31 07:05:52.000000000 -0600
+++ singular-4.1.1/configure.ac	2018-10-24 08:33:42.979724820 -0600
@@ -264,4 +264,10 @@ AC_CONFIG_FILES([redhat/Makefile])
 AC_CONFIG_FILES([redhat/singular.spec])
 AC_CONFIG_FILES([desktop/Makefile desktop/Singular.desktop desktop/Singular-manual.desktop])
 
+AC_CONFIG_COMMANDS([norpath], [
+sed -e 's|^hardcode_libdir_flag_spec=.*|hardcode_libdir_flag_spec=""|g' \
+    -e 's|^runpath_var=LD_RUN_PATH|runpath_var=DIE_RPATH_DIE|g' \
+    -e 's|CC="\(g.*\)"|CC="\1 -Wl,--as-needed"|' \
+    -i libtool
+])
 AC_OUTPUT
--- singular-4.1.1/factory/configure.ac.orig	2018-07-31 07:07:38.000000000 -0600
+++ singular-4.1.1/factory/configure.ac	2018-10-24 08:33:42.980724804 -0600
@@ -412,5 +412,11 @@ AC_DEFINE_UNQUOTED([FACTORYCONFIGURATION
 
 
 AC_CONFIG_FILES([Makefile include/factory/Makefile factory.pc]) #  ftest/GNUmakefile
+AC_CONFIG_COMMANDS([norpath], [
+sed -e 's|^hardcode_libdir_flag_spec=.*|hardcode_libdir_flag_spec=""|g' \
+    -e 's|^runpath_var=LD_RUN_PATH|runpath_var=DIE_RPATH_DIE|g' \
+    -e 's|CC="\(g.*\)"|CC="\1 -Wl,--as-needed"|' \
+    -i libtool
+])
 AC_OUTPUT
 
--- singular-4.1.1/gfanlib/configure.ac.orig	2018-07-31 07:07:38.000000000 -0600
+++ singular-4.1.1/gfanlib/configure.ac	2018-10-24 08:33:42.981724789 -0600
@@ -57,4 +57,10 @@ AC_LANG_POP()
 
 AC_CONFIG_FILES([Makefile])
 ## AC_CONFIG_HEADERS([config.h])
+AC_CONFIG_COMMANDS([norpath], [
+sed -e 's|^hardcode_libdir_flag_spec=.*|hardcode_libdir_flag_spec=""|g' \
+    -e 's|^runpath_var=LD_RUN_PATH|runpath_var=DIE_RPATH_DIE|g' \
+    -e 's|CC="\(g.*\)"|CC="\1 -Wl,--as-needed"|' \
+    -i libtool
+])
 AC_OUTPUT
--- singular-4.1.1/gfanlib/Makefile.am.orig	2018-07-31 07:07:38.000000000 -0600
+++ singular-4.1.1/gfanlib/Makefile.am	2018-10-24 08:33:42.981724789 -0600
@@ -2,6 +2,7 @@ ACLOCAL_AMFLAGS = -I ../m4
 
 if HAVE_GFANLIB
   libgfan_LTLIBRARIES=libgfan.la
+  libgfan_la_LIBADD=-lcddgmp -lgmp
 endif
 
 libgfandir = $(libdir)
--- singular-4.1.1/libpolys/configure.ac.orig	2018-07-31 07:07:38.000000000 -0600
+++ singular-4.1.1/libpolys/configure.ac	2018-10-24 08:33:42.982724773 -0600
@@ -159,4 +159,10 @@ AC_CONFIG_FILES([coeffs/Makefile])
 AC_CONFIG_FILES([polys/Makefile])
 AC_CONFIG_FILES([tests/Makefile])
 AC_CONFIG_FILES([libpolys-config libpolys.pc])
+AC_CONFIG_COMMANDS([norpath], [
+sed -e 's|^hardcode_libdir_flag_spec=.*|hardcode_libdir_flag_spec=""|g' \
+    -e 's|^runpath_var=LD_RUN_PATH|runpath_var=DIE_RPATH_DIE|g' \
+    -e 's|CC="\(g.*\)"|CC="\1 -Wl,--as-needed"|' \
+    -i libtool
+])
 AC_OUTPUT
--- singular-4.1.1/omalloc/configure.ac.orig	2018-07-31 07:07:38.000000000 -0600
+++ singular-4.1.1/omalloc/configure.ac	2018-10-24 08:33:42.982724773 -0600
@@ -600,5 +600,11 @@ dnl llllllllllllllllllllllllllllllllllll
 dnl wrap it up
 dnl
 AC_CONFIG_FILES([Makefile omalloc.pc])
+AC_CONFIG_COMMANDS([norpath], [
+sed -e 's|^hardcode_libdir_flag_spec=.*|hardcode_libdir_flag_spec=""|g' \
+    -e 's|^runpath_var=LD_RUN_PATH|runpath_var=DIE_RPATH_DIE|g' \
+    -e 's|CC="\(g.*\)"|CC="\1 -Wl,--as-needed"|' \
+    -i libtool
+])
 AC_OUTPUT
 
--- singular-4.1.1/resources/configure.ac.orig	2018-07-31 07:07:38.000000000 -0600
+++ singular-4.1.1/resources/configure.ac	2018-10-24 08:33:42.983724758 -0600
@@ -74,4 +74,10 @@ AX_NORMALIZE_PATH([config_datadir],['/']
 AC_DEFINE_UNQUOTED([DATA_DIR],"$config_datadir",[datadir])
 
 AC_CONFIG_FILES([singular_resources.pc Makefile])
+AC_CONFIG_COMMANDS([norpath], [
+sed -e 's|^hardcode_libdir_flag_spec=.*|hardcode_libdir_flag_spec=""|g' \
+    -e 's|^runpath_var=LD_RUN_PATH|runpath_var=DIE_RPATH_DIE|g' \
+    -e 's|CC="\(g.*\)"|CC="\1 -Wl,--as-needed"|' \
+    -i libtool
+])
 AC_OUTPUT
