diff -up Singular-3-1-7/doc/doc2tex.pl.orig Singular-3-1-7/doc/doc2tex.pl
--- Singular-3-1-7/doc/doc2tex.pl.orig	2014-08-06 09:59:15.000000000 -0600
+++ Singular-3-1-7/doc/doc2tex.pl	2016-02-18 21:23:21.197141422 -0700
@@ -479,7 +479,7 @@ sub HandleRef
     $line++;
     last if (/^\@c\s*ref\s*$/);
     
-    while (/\@ref{(.*?)}[;\.]/)
+    while (/\@ref\{(.*?)\}[;\.]/)
     {
       $refs{$1} = 1;
       $_ = $';
diff -up Singular-3-1-7/doc/texi2html.orig Singular-3-1-7/doc/texi2html
--- Singular-3-1-7/doc/texi2html.orig	2014-08-06 09:59:15.000000000 -0600
+++ Singular-3-1-7/doc/texi2html	2016-02-18 21:26:21.432652047 -0700
@@ -3063,7 +3063,7 @@ INPUT_LINE: while ($_ = &next_line) {
     #############################################################
     # value substitution before macro expansion, so that
     # it works in macro arguments
-    s/\@value{($VARRE)}/$value{$1}/eg;
+    s/\@value\{($VARRE)\}/$value{$1}/eg;
 
     #############################################################
     # macro substitution
@@ -3089,8 +3089,8 @@ INPUT_LINE: while ($_ = &next_line) {
 	  $after = '';
 	}
 	$args =~ s|\\\\|\\|g;
-	$args =~ s|\\{|{|g;
-        $args =~ s|\\}|}|g;
+	$args =~ s|\\\{|{|g;
+        $args =~ s|\\\}|}|g;
 	if (@{$macros->{$name}->{Args}} > 1)
 	{
 	  $args =~ s/(^|[^\\]),/$1$;/g ;
@@ -4243,7 +4243,7 @@ while (@lines) {
     #
     # xref
     #
-    while (/\@(x|px|info|)ref{([^{}]+)(}?)/) {
+    while (/\@(x|px|info|)ref\{([^{}]+)(\}?)/) {
 	# note: Texinfo may accept other characters
 	($type, $nodes, $full) = ($1, $2, $3);
 	($before, $after) = ($`, $');
@@ -4898,9 +4898,9 @@ sub update_sec_num {
     my $ret;
 
     $level--; # here we start at 0
-    if ($name =~ /^appendix/ || defined(@appendix_sec_num)) {
+    if ($name =~ /^appendix/ || @appendix_sec_num) {
 	# appendix style
-	if (defined(@appendix_sec_num)) {
+	if (@appendix_sec_num) {
 	    &incr_sec_num($level, @appendix_sec_num);
 	} else {
 	    @appendix_sec_num = ('A', 0, 0, 0);
@@ -4908,7 +4908,7 @@ sub update_sec_num {
 	$ret = join('.', @appendix_sec_num[0..$level]);
     } else {
 	# normal style
-	if (defined(@normal_sec_num)) 
+	if (@normal_sec_num)
 	{
 	  &incr_sec_num($level, @normal_sec_num);
 	} 
