diff -up Singular-3-1-3/Singular/configure.orig Singular-3-1-3/Singular/configure
--- Singular-3-1-3/Singular/configure.orig	2012-05-05 14:09:57.443956412 -0300
+++ Singular-3-1-3/Singular/configure	2012-05-05 14:11:06.261959072 -0300
@@ -1800,7 +1800,7 @@ python_exe_name="no"
 PYTHON_PLUGIN="module"
 PYTHON_LINKAGE="static"
 
-if test "${with_python+set}" == set; then
+if test "${with_python+set}" = set; then
   case "${with_python}" in
     yes|static*|dynamic*|shared*|module*|embed*)
       python_exe_name="python"
@@ -1971,7 +1971,7 @@ echo "configure:1968: checking for Pytho
 		py_version=`$PYTHON -c "from distutils.sysconfig import *; \
                         from string import join; \
                         print join(get_config_vars('VERSION'))"`
-		if test "$py_version" == "None"; then
+		if test "$py_version" = "None"; then
 			if test -n "$PYTHON_VERSION"; then
 				py_version=$PYTHON_VERSION
 			else
@@ -2134,7 +2134,7 @@ rm -f conftest*
 
         echo "$ac_t""$pythonstaticexists" 1>&6
 
-        if test "$pythonstaticexists" == yes ; then
+        if test "$pythonstaticexists" = yes ; then
           cat >> confdefs.h <<\EOF
 #define HAVE_STATIC_PYTHON 1
 EOF
