diff -up Singular-3-1-7/gfanlib/gfanlib_matrix.h.orig Singular-3-1-7/gfanlib/gfanlib_matrix.h
--- Singular-3-1-7/gfanlib/gfanlib_matrix.h.orig	2014-08-06 09:59:15.000000000 -0600
+++ Singular-3-1-7/gfanlib/gfanlib_matrix.h	2016-02-18 20:02:45.235928428 -0700
@@ -327,7 +327,7 @@ public:
               }
             if(makePivotsOne)
               {//THE PIVOT SHOULD BE SET TO ONE IF INTEGRAL IS FALSE
-                if(!rows[currentRow][i].sign()<0)retSwaps++;
+                if(rows[currentRow][i].sign()>=0)retSwaps++;
                 typ inverse=typ(1)/rows[currentRow][i];
                 //                if(!rows[currentRow][i].isOne())
                   {
