diff -up Singular-3-1-5/Singular/Makefile.in.flint24 Singular-3-1-5/Singular/Makefile.in
--- Singular-3-1-5/Singular/Makefile.in.flint24	2014-03-10 13:48:54.812129117 -0500
+++ Singular-3-1-5/Singular/Makefile.in	2014-03-10 13:50:12.069307410 -0500
@@ -132,7 +132,7 @@ HAVE_MPSR       = @HAVE_MPSR@
 #
 # Handle libSINGULAR stuff
 #
-LIBSINGULAR_LIBS =-lsingfac -lsingcf -lntl -lreadline -lgmp -lomalloc -lpthread -ldl
+LIBSINGULAR_LIBS =-lflint -lsingfac -lsingcf -lntl -lreadline -lgmp -lomalloc -lpthread -ldl
 
 SO_SUFFIX = so
 
