--- singular-4.1.0/configure.ac.orig	2017-02-20 10:33:27.000000000 -0700
+++ singular-4.1.0/configure.ac	2017-02-27 14:49:09.415812935 -0700
@@ -253,4 +253,10 @@ AC_CONFIG_FILES([redhat/Makefile])
 AC_CONFIG_FILES([redhat/singular.spec])
 AC_CONFIG_FILES([desktop/Makefile desktop/Singular.desktop desktop/Singular-manual.desktop])
 
+AC_CONFIG_COMMANDS([norpath], [
+sed -e 's|^hardcode_libdir_flag_spec=.*|hardcode_libdir_flag_spec=""|g' \
+    -e 's|^runpath_var=LD_RUN_PATH|runpath_var=DIE_RPATH_DIE|g' \
+    -e 's|CC="\(g.*\)"|CC="\1 -Wl,--as-needed"|' \
+    -i libtool
+])
 AC_OUTPUT
--- singular-4.1.0/factory/configure.ac.orig	2017-02-20 10:34:57.000000000 -0700
+++ singular-4.1.0/factory/configure.ac	2017-02-27 14:49:09.416812931 -0700
@@ -407,5 +407,11 @@ AC_DEFINE_UNQUOTED([FACTORYCONFIGURATION
 
 
 AC_CONFIG_FILES([Makefile include/factory/Makefile factory.pc]) #  ftest/GNUmakefile
+AC_CONFIG_COMMANDS([norpath], [
+sed -e 's|^hardcode_libdir_flag_spec=.*|hardcode_libdir_flag_spec=""|g' \
+    -e 's|^runpath_var=LD_RUN_PATH|runpath_var=DIE_RPATH_DIE|g' \
+    -e 's|CC="\(g.*\)"|CC="\1 -Wl,--as-needed"|' \
+    -i libtool
+])
 AC_OUTPUT
 
--- singular-4.1.0/gfanlib/configure.ac.orig	2017-02-20 10:34:57.000000000 -0700
+++ singular-4.1.0/gfanlib/configure.ac	2017-02-27 14:49:09.416812931 -0700
@@ -57,4 +57,10 @@ AC_LANG_POP()
 
 AC_CONFIG_FILES([Makefile])
 ## AC_CONFIG_HEADERS([config.h])
+AC_CONFIG_COMMANDS([norpath], [
+sed -e 's|^hardcode_libdir_flag_spec=.*|hardcode_libdir_flag_spec=""|g' \
+    -e 's|^runpath_var=LD_RUN_PATH|runpath_var=DIE_RPATH_DIE|g' \
+    -e 's|CC="\(g.*\)"|CC="\1 -Wl,--as-needed"|' \
+    -i libtool
+])
 AC_OUTPUT
--- singular-4.1.0/gfanlib/Makefile.am.orig	2017-02-20 10:34:57.000000000 -0700
+++ singular-4.1.0/gfanlib/Makefile.am	2017-02-27 14:49:09.416812931 -0700
@@ -2,6 +2,7 @@ ACLOCAL_AMFLAGS = -I ../m4
 
 if HAVE_GFANLIB
   libgfan_LTLIBRARIES=libgfan.la
+  libgfan_la_LIBADD=-lcddgmp -lgmp
 endif
 
 libgfandir = $(libdir)
--- singular-4.1.0/libpolys/configure.ac.orig	2017-02-20 10:34:57.000000000 -0700
+++ singular-4.1.0/libpolys/configure.ac	2017-02-27 14:49:09.416812931 -0700
@@ -149,4 +149,10 @@ AC_CONFIG_FILES([coeffs/Makefile])
 AC_CONFIG_FILES([polys/Makefile])
 AC_CONFIG_FILES([tests/Makefile])
 AC_CONFIG_FILES([libpolys-config libpolys.pc])
+AC_CONFIG_COMMANDS([norpath], [
+sed -e 's|^hardcode_libdir_flag_spec=.*|hardcode_libdir_flag_spec=""|g' \
+    -e 's|^runpath_var=LD_RUN_PATH|runpath_var=DIE_RPATH_DIE|g' \
+    -e 's|CC="\(g.*\)"|CC="\1 -Wl,--as-needed"|' \
+    -i libtool
+])
 AC_OUTPUT
--- singular-4.1.0/omalloc/configure.ac.orig	2017-02-20 10:34:57.000000000 -0700
+++ singular-4.1.0/omalloc/configure.ac	2017-02-27 14:49:09.416812931 -0700
@@ -636,5 +636,11 @@ dnl llllllllllllllllllllllllllllllllllll
 dnl wrap it up
 dnl
 AC_CONFIG_FILES([Makefile omalloc.pc])
+AC_CONFIG_COMMANDS([norpath], [
+sed -e 's|^hardcode_libdir_flag_spec=.*|hardcode_libdir_flag_spec=""|g' \
+    -e 's|^runpath_var=LD_RUN_PATH|runpath_var=DIE_RPATH_DIE|g' \
+    -e 's|CC="\(g.*\)"|CC="\1 -Wl,--as-needed"|' \
+    -i libtool
+])
 AC_OUTPUT
 
--- singular-4.1.0/resources/configure.ac.orig	2017-02-20 10:34:57.000000000 -0700
+++ singular-4.1.0/resources/configure.ac	2017-02-27 14:49:09.416812931 -0700
@@ -73,4 +73,10 @@ AX_NORMALIZE_PATH([config_datadir],['/']
 AC_DEFINE_UNQUOTED([DATA_DIR],"$config_datadir",[datadir])
 
 AC_CONFIG_FILES([singular_resources.pc Makefile])
+AC_CONFIG_COMMANDS([norpath], [
+sed -e 's|^hardcode_libdir_flag_spec=.*|hardcode_libdir_flag_spec=""|g' \
+    -e 's|^runpath_var=LD_RUN_PATH|runpath_var=DIE_RPATH_DIE|g' \
+    -e 's|CC="\(g.*\)"|CC="\1 -Wl,--as-needed"|' \
+    -i libtool
+])
 AC_OUTPUT
