--- Singular-3-1-6/factory/NTLconvert.cc.orig	2012-12-19 14:01:16.000000000 -0700
+++ Singular-3-1-6/factory/NTLconvert.cc	2014-06-17 12:30:00.000000000 -0600
@@ -55,6 +55,23 @@
 NTL_CLIENT
 #endif
 
+// Needed explicit template instantiations for NTL 6.x.x
+template mat_zz_p::Mat(const mat_zz_p&);
+template mat_zz_pE::Mat(const mat_zz_pE&);
+
+template class Vec<Pair<GF2X, long> >;
+template class Vec<Pair<GF2EX, long> >;
+template class Vec<Pair<zz_pX, long> >;
+template class Vec<Pair<zz_pEX, long> >;
+template class Vec<Pair<ZZ_pX, long> >;
+template class Vec<Pair<ZZ_pEX, long> >;
+template class Vec<Pair<ZZX, long> >;
+template class Vec<Vec<zz_p> >;
+template class Vec<Vec<zz_pE> >;
+
+template long operator==<zz_p>(const vec_zz_p&, const vec_zz_p&);
+template Vec<zz_p>::~Vec();
+
 ////////////////////////////////////////////////////////////////////////////////
 // NAME: convertFacCF2NTLZZpX                                                 //
 //                                                                            //
