diff -up Singular-3-1-3/kernel/Makefile.in.orig Singular-3-1-3/kernel/Makefile.in
--- Singular-3-1-3/kernel/Makefile.in.orig	2012-05-05 15:03:36.064079528 -0300
+++ Singular-3-1-3/kernel/Makefile.in	2012-05-05 15:03:55.870080289 -0300
@@ -48,7 +48,7 @@ PIPE		= @PIPE@
 CFLAGS		= @CFLAGS@ ${PIPE} 
 CXXFLAGS	= @CXXFLAGS@ ${PIPE} 
 CXXTEMPLFLAGS	= @CXXTEMPLFLAGS@
-CPPFLAGS	= -I${srcdir} -I.. -I@prefix@  @CPPFLAGS@ 
+CPPFLAGS	= -I${srcdir} -I.. -I../omalloc -I${srcdir}/../factory -I@prefix@  @CPPFLAGS@ 
 DEFS		= -DNDEBUG -DOM_NDEBUG -D@SING_UNAME@ @DEFS@
 LDFLAGS		= @LDFLAGS@
 LD_DYN_FLAGS	= @LD_DYN_FLAGS@
diff -up Singular-3-1-3/libfac/Makefile.in.orig Singular-3-1-3/libfac/Makefile.in
diff -up Singular-3-1-3/libfac/Makefile.in.orig Singular-3-1-3/libfac/Makefile.in
--- Singular-3-1-3/libfac/Makefile.in.orig	2012-05-05 15:16:31.642109243 -0300
+++ Singular-3-1-3/libfac/Makefile.in	2012-05-05 15:17:01.494110342 -0300
@@ -37,7 +37,7 @@ MKINSTALLDIRS	= ./mkinstalldirs
 ##
 CXXFLAGS	= @CXXFLAGS@
 CXXTEMPLFLAGS	= @CXXTEMPLFLAGS@
-CPPFLAGS	= -I${srcdir}/factor -I${srcdir}/charset -I${srcdir} -I.. -I${prefix} @CPPFLAGS@ 
+CPPFLAGS	= -I${srcdir}/factor -I${srcdir}/charset -I${srcdir} -I${srcdir}/.. -I${srcdir}/../factory -I${prefix} @CPPFLAGS@ 
 DEFS		= @DEFS@ 
 LDFLAGS		= @LDFLAGS@
 LIBS		= -lfac -lcf -lcfmem -lgmp @LIBS@
diff -up Singular-3-1-3/Singular/Makefile.in.orig Singular-3-1-3/Singular/Makefile.in
--- Singular-3-1-3/Singular/Makefile.in.orig	2012-05-05 15:25:21.273129522 -0300
+++ Singular-3-1-3/Singular/Makefile.in	2012-05-05 15:26:08.502131306 -0300
@@ -76,7 +76,7 @@ CFLAGS		= @CFLAGS@ ${PIPE}
 CXXFLAGS	= @CXXFLAGS@ -I.. -I@prefix@ ${PIPE}
 CXXTEMPLFLAGS	= @CXXTEMPLFLAGS@ -I.. -I@prefix@
 ifneq ($(SINGUNAME),ix86-Win)
-CPPFLAGS	= -I${srcdir} -I.. -I@prefix@ @CPPFLAGS@ @PYTHON_CPPFLAGS@
+CPPFLAGS	= -I${srcdir} -I${srcdir}/.. -I${srcdir}/../omalloc -I@prefix@ @CPPFLAGS@ @PYTHON_CPPFLAGS@
 else
 ## Dirs after @CPPFLAGS@ are for PySingular
 CPPFLAGS	= -I${srcdir} -I.. -I@prefix@ @CPPFLAGS@ @PYTHON_CPPFLAGS@
